### ISSUE TITLE
[Suggestion] Use SGB Borders with GBC Colors in Dual-Mode GBC Games

### ISSUE THREAD

**Issue by PW5190** [mgba-emu/mgba#2832]
I have a suggestion. I've been playing Pokémon Yellow, Gold, and Silver in Pokémon Stadium 2 on my Nintendo 64 via the GB Tower. These games show their special Super Game Boy borders, but display with colorful graphics, as if being played on a Game Boy Color.

I've been thinking... What if we can implement this feature for every dual-mode Game Boy Color game, displaying SGB borders, while staying true to the GBC colors in-game? We know certain caveats for dual-mode GBC games being played on SGB, while the SGB *was* capable of playing Game Boy games in color, the GBC displays *more* colors than the SGB, but does not feature SGB enhancements, such as the special borders.

Any thoughts of this idea so far?

**Comment by endrift** [comment]
It's not possible to do this without hacks. That said, lots of GB emulators have these hacks and I was planning to add some way of handling this in mGBA.

**Comment by PW5190** [comment]
> It's not possible to do this without hacks. That said, lots of GB emulators have these hacks and I was planning to add some way of handling this in mGBA.

I understand. It can be quite difficult to implement, but as far as I can tell, Pokémon Yellow, Gold, Silver when running within Pokémon Stadium 2's GB Tower play in GBC mode, but display their SGB borders anyway. Pretty cool, I must say.

Anyways, back on track. If we could allow mGBA to do the same, in theory, we could allow dual-mode GBC games to run in GBC mode and display their SGB borders, it can be faithful to Pokémon Stadium 2's GB Tower mode!

**Comment by endrift** [comment]
Yup, that's on the roadmap. I was going to do it in 0.11 but that's been delayed for so long at this point that I just wanna get it out the door and add more features afterwards

**Comment by Squall-Leonhart** [comment]
In CGB mode, the packets for SGB borders just never appear so theres going to be a limit for what games actually work in this situation without doing awkward resets - most emulators (and even Pokemon Stadium 2s GB Tower) start off as super gameboy and reset to CGB

Dragon Warriors for instance doesn't do borders until after the naming screen, or loading a save game, and then changes depending on if you're in town, the overworld or the castle - the half baked method in VBA-M currently runs as SGB until it reaches any of these then reboots into Color with the given border for the location specified - and never updates again because theres no further border uploads - This will be fixed by preferring CGB mode over loading a border on these cases, something which GB Tower seems to perform too, but...

I have a local branch of VBAM emulating both cores in sync and performing border updates from the the SGB state while only presenting video and audio of the CGB state, it...works.... but theres lots of state duplication.
